NUI/Core/NUISettings.m

```diff
@@ -218,14 +218,33 @@ + (UIFont*)getFontWithClass:(NSString*)className baseFont:(UIFont *)baseFont
     if ([self hasProperty:propertyName withClass:className]) {
         NSString *fontName = [self get:propertyName withClass:className];
         
-        if ([fontName isEqualToString:@"system"]) {
-            font = [UIFont systemFontOfSize:fontSize];
-        } else if ([fontName isEqualToString:@"boldSystem"]) {
-            font = [UIFont boldSystemFontOfSize:fontSize];
-        } else if ([fontName isEqualToString:@"italicSystem"]) {
-            font = [UIFont italicSystemFontOfSize:fontSize];
-        } else {
-            font = [UIFont fontWithName:fontName size:fontSize];
+        if ([[UIFont class] respondsToSelector:@selector(systemFontOfSize:weight:)]) {
+            if ([fontName isEqualToString:@"blackSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightBlack];
+            } else if ([fontName isEqualToString:@"heavySystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightHeavy];
+            } else if ([fontName isEqualToString:@"lightSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightLight];
+            } else if ([fontName isEqualToString:@"mediumSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightMedium];
+            } else if ([fontName isEqualToString:@"semiboldSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightSemibold];
+            } else if ([fontName isEqualToString:@"thinSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightThin];
+            } else if ([fontName isEqualToString:@"ultraLightSystem"]) {
+                font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightUltraLight];
+            }
+        }
+        if (!font) {
```

Comment:
Just something that came to my mind (wouldn't make any effort to change it):
Adding this check to the last else block would require less nesting